src/mqueryfront/src/StorageList.js

```diff
@@ -0,0 +1,115 @@
+import React, { Component } from "react";
+import ErrorBoundary from "./ErrorBoundary";
+
+class StorageRow extends Component {
+    getWatchedBadge() {
+        if (this.props.enabled) {
+            return (
+                <span
+                    class="badge badge-success"
+                    data-toggle="tooltip"
+                    title="This location is watched and indexed automatically"
+                >
+                    watched
+                </span>
+            );
+        }
+        return (
+            <span
+                class="badge badge-secondary"
+                data-toggle="tooltip"
+                title="This location is not indexed automatically"
+            >
+                disabled
+            </span>
+        );
+    }
+
+    render() {
+        var taintTags = this.props.taints.map((taint) => (
+            <span>
+                {" "}
+                <span
+                    class="badge badge-primary"
+                    data-toggle="tooltip"
+                    title={`New indexed files are tagged with "${taint}"`}
+                >
+                    {taint}
+                </span>
+            </span>
+        ));
+
+        var actionButtons = (
+            <div>
+                <button
+                    type="button"
+                    class="btn btn-secondary btn-sm"
+                    data-toggle="tooltip"
+                    title="Reindex this dataset now"
+                >
+                    Reindex
+                </button>{" "}
+                <button
+                    type="button"
+                    class="btn btn-danger btn-sm"
+                    data-toggle="tooltip"
+                    title="Stop watching the dataset (keep indexed files)"
+                >
+                    Delete
+                </button>
+            </div>
+        );
+
+        return (
+            <tr>
+                <td>
+                    <code>{this.props.name}</code>
+                    {taintTags}
+                </td>
+                <td>
+                    <code>{this.props.path}</code>
+                </td>
+                <td>
+                    {new Date(this.props.lastUpdate).toLocaleDateString()}{" "}
+                    {this.getWatchedBadge()}
```

Comment:
looks like a good place to do string interpolation, but ok

Reply:
hmm, these are all DOM elements so (unless i misunderstand you) it's not possible